OPAL/tac/src/main/scala/org/opalj/tac/cg/android/AndroidICCAnalysis.scala

```diff
@@ -0,0 +1,1467 @@
+    )
+    val broadcastReceiverStartMethods: List[String] = List(
+        "sendBroadcast", "sendOrderedBroadcast", "sendStickyBroadcast", "sendStickyOrderedBroadcast", "getBroadcast",
+        "sendBroadcastAsUser", "sendStickyBroadcastAsUser", "sendStickyOrderedBroadcastAsUser"
+    )
+
+    final val intentMethods = List(
+        setAction, addCategory, setData, setDataAndNormalize, setDataAndType, setDataAndTypeAndNormalize, setType, setTypeAndNormalize, setPackage
+    )
+    final val explicitMethods = List(setClass, setClassName, setComponent)
+
+    val intentFilters: ListBuffer[IntentFilter] = ListBuffer.empty[IntentFilter]
+    val edges: ListBuffer[((Method, Int), Method)] = ListBuffer.empty[((Method, Int), Method)]
+    val explicitIntents: ListBuffer[ExplicitIntent] = ListBuffer.empty[ExplicitIntent]
+    val implicitIntents: ListBuffer[ImplicitIntent] = ListBuffer.empty[ImplicitIntent]
+
+    val senderMap: mutable.Map[(ObjectType, String), ListBuffer[(Method, String, Int)]] =
+        mutable.Map.empty[(ObjectType, String), ListBuffer[(Method, String, Int)]] //Class and name of the Method that returns the searched intent -> the intent sending methods and the component type
+    val returningIntentsMap: mutable.Map[(ObjectType, String), ListBuffer[ImplicitIntent]] =
+        mutable.Map.empty[(ObjectType, String), ListBuffer[ImplicitIntent]] //class and method that returns an intent -> intent
+
+    var leftOverMethods: Set[Method] = Set.empty[Method]
+    var notStarted = true
+
+    val source: BufferedSource = Source.fromInputStream(this.getClass.getResourceAsStream("AndroidCallbackList.txt"))
+    val callbackList: Set[ObjectType] = source.getLines.map(ObjectType(_)).toSet
+    source.close
+
+    /**
+     * Calls all necessary methods to generate the call graph edges that are Android specific.
+     *
+     * @param manifest AndroidManifest.xml
+     */
+    def performAnalysis(
+        project: SomeProject
+    ): ProperPropertyComputationResult = {
+        val manifestParsingResultOption = project.get(AndroidManifestKey)
+        if (manifestParsingResultOption.isDefined) {
+            val manifestParsingResult = manifestParsingResultOption.get
+            intentFilters ++= manifestParsingResult._2
+            generateLifecycleCallbacks(manifestParsingResult._1)
+        }
+        searchIntentsAndCallbacks()
+    }
+
+    /**
+     * Starts the second phase of the analysis, after all methods are processed once to collect information on intents
+     * and intent filters.
+     *
+     * @param project The analysed project.
+     * @return Returns the calls originating from inter-component-communication.
+     */
+    def startSecondPhase(project: SomeProject): ProperPropertyComputationResult = {
+        intentMatching()
+
+        val decMeths = project.get(DeclaredMethodsKey)
+        val contexts = project.get(SimpleContextsKey)
+        val resList = ListBuffer.empty[PartialResult[_, _ >: Null <: Property]]
```

Comment:
Do not use a (mutable) data structure here, just declare this as `TraversableOnce` to ensure that any iterators (in `partialResults` are not evaluated but kept lazy.